### PR TITLE
Float types support

### LINKED ISSUE #1353: f32 or f64 arent supported in anchor idl
I have defined a field of type floating point in #[account], then run command "anchor test", got a error.

**IdlError: Type not found: {"name":"nums","type":{"defined":"f32"}}**
    at Function.fieldLayout (/mnt/d/it/work/blockchain/token-faucet/node_modules/@project-serum/anchor/src/coder/idl.ts:88:19)
    at /mnt/d/it/work/blockchain/token-faucet/node_modules/@project-serum/anchor/src/coder/idl.ts:116:28
    at Array.map (<anonymous>)
    at Function.typeDefLayout (/mnt/d/it/work/blockchain/token-faucet/node_modules/@project-serum/anchor/src/coder/idl.ts:115:48)
    at /mnt/d/it/work/blockchain/token-faucet/node_modules/@project-serum/anchor/src/coder/accounts.ts:28:39
    at Array.map (<anonymous>)

### PR DESCRIPTION
**Pull request by pashpashkin**
Closes https://github.com/project-serum/anchor/issues/1353

### CODE REVIEW

File: tests/floats/tests/floats.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+describe("floats", () => {
+  // Configure the client to use the local cluster.
+  anchor.setProvider(anchor.Provider.env());
+
+  const program = anchor.workspace.Floats as Program<Floats>;
+
+  it("Creates an account with float data", async () => {
+    const accountKeypair = anchor.web3.Keypair.generate();
+
+    await program.rpc.create(1.0, 2.0, {
```
Comment (armaniferrante) [inline tests/floats/tests/floats.ts:15]: Please change all new tests to use the more modern `program.methods` API. Other namespaces will eventually be deprecated.

**Comment by pashpashkin** [comment]
@armaniferrante, could you take a look if anything else can be improved in this PR or it's ready to be merged?

**Comment by armaniferrante** [comment]
Please add a CHANGELOG.md entry and this is good to go!

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ incremented for features.
 ### Features
 
 * lang: add check that declared id == program id ([#1451](https://github.com/project-serum/anchor/pull/1451))
+* ts: Added float types support ([#1425](https://github.com/project-serum/anchor/pull/1425))
```
Comment (armaniferrante) [inline CHANGELOG.md:17]:
```suggestion
* ts: Added float types support ([#1425](https://github.com/project-serum/anchor/pull/1425)).
```

**Pull request merged**